arch/arm/soc/atmel_sam/common/soc_gpio.c

```diff
@@ -44,10 +47,22 @@ static void configure_input_attr(Pio *pio, u32_t mask, u32_t flags)
 	if ((flags & SOC_GPIO_IN_FILTER_MASK) != 0) {
 		if ((flags & SOC_GPIO_IN_FILTER_MASK) == SOC_GPIO_IN_FILTER_DEBOUNCE) {
 			/* Enable de-bounce, disable de-glitch */
+#if defined PIO_IFSCER_P0
 			pio->PIO_IFSCER = mask;
+#elif defined PIO_DIFSR_P0
+			pio->PIO_DIFSR = mask;
+#else
+#error "Unsupported Atmel SAM MCU series"
```

Comment:
Can't you just do that check at the beginning of the file, once, instead? (if !defined(...) && !defined(...) #error "..." #endif )
Here it's repeating #error and makes the code quite ugly.